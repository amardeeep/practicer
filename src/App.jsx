import './App.css'
import { useState } from 'react'

function App() {
    const [heading,setHeading]=useState("Banana")
    function clickHandler(){
        setHeading("Mango")
    }
 return(
    <div>
        <button onClick={clickHandler}>Change Heading !</button>
    <h1>{heading}</h1>
    </div>
 )
}

export default App
