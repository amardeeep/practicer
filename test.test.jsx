import { render, screen } from "@testing-library/react";

import App from "./src/App";
import { describe, it, expect } from "vitest";
import userEvent from "@testing-library/user-event";

describe("App Component", () => {
  it("It renders Banana", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("it renders Mango after button click", async () => {
    const user = userEvent.setup();
    render(<App />);
    const button = screen.getByRole("button");
    await user.click(button);
    expect(screen.getByRole("heading").textContent).toMatch(/mango/i);
  });
});

/*describe('App', () => {
  it('renders headline', () => {
    render(<App title="React" />);

    screen.debug();

    // check if App components renders headline
  });
});*/
