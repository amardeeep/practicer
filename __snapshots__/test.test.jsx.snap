// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > It renders Banana 1`] = `
<div>
  <div>
    <button>
      Change Heading !
    </button>
    <h1>
      Banana
    </h1>
  </div>
</div>
`;
